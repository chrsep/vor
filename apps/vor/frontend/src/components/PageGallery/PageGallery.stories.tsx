import React, { FC } from "react"
import PageGallery from "./PageGallery"

export default {
  title: "Core|PageGallery",
  component: PageGallery,
  parameters: {
    componentSubtitle: "Just a simple PageGallery",
  },
}

export const Basic: FC = () => <PageGallery id="asdasd" />
