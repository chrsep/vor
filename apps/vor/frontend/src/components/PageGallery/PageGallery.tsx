/** @jsx jsx * */
import { keyframes } from "@emotion/react"
import { Trans } from "@lingui/macro"
import { Suspense, ChangeEvent, FC, Fragment, useState } from "react"
import { Box, Button, Flex, Image, Label, jsx } from "theme-ui"
import { getFirstName } from "../../domain/person"
import { useUploadStudentVideo } from "../../hooks/api/schools/useUploadStudentVideo"
import useGetStudentImages, {
  StudentImage,
} from "../../hooks/api/students/useGetStudentImages"
import useGetVideos from "../../hooks/api/students/useGetVideos"
import usePostNewStudentImage from "../../hooks/api/students/usePostNewStudentImage"
import { useGetStudent } from "../../hooks/api/useGetStudent"
import useVisibilityState from "../../hooks/useVisibilityState"
import { ReactComponent as PlusIcon } from "../../icons/plus.svg"
import {
  STUDENT_IMAGE_URL,
  STUDENT_OVERVIEW_URL,
  STUDENTS_URL,
} from "../../routes"
import Dialog from "../Dialog/Dialog"
import Icon from "../Icon/Icon"
import { Link } from "../Link/Link"
import LoadingIndicator from "../LoadingIndicator/LoadingIndicator"
import LoadingPlaceholder from "../LoadingPlaceholder/LoadingPlaceholder"
import Tab from "../Tab/Tab"
import TopBar, { breadCrumb } from "../TopBar/TopBar"
import TranslucentBar from "../TranslucentBar/TranslucentBar"
import { Typography } from "../Typography/Typography"
import LazyVideoPlayer from "../VideoPlayer/LazyVideoPlayer"
import { ReactComponent as CloseIcon } from "../../icons/close.svg"

export interface PageGalleryProps {
  studentId: string
}

enum View {
  IMAGES,
  VIDEOS,
}

const PageGallery: FC<PageGalleryProps> = ({ studentId }) => {
  const student = useGetStudent(studentId)
  const [selectedView, setSelectedView] = useState(View.IMAGES)

  return (
    <Box>
      <TranslucentBar boxSx={{ position: "sticky", top: -48, zIndex: 10 }}>
        <TopBar
          breadcrumbs={[
            breadCrumb("Students", STUDENTS_URL),
            breadCrumb(
              getFirstName(student.data),
              STUDENT_OVERVIEW_URL(studentId)
            ),
            breadCrumb("Gallery"),
          ]}
        />

        <Tab
          items={["Images", "Videos"]}
          selectedItemIdx={selectedView}
          onTabClick={(idx) => {
            setSelectedView(idx)
          }}
        />
      </TranslucentBar>

      {selectedView === View.IMAGES && <ImagesView studentId={studentId} />}
      {selectedView === View.VIDEOS && <VideosView studentId={studentId} />}
    </Box>
  )
}

const ImagesView: FC<{ studentId: string }> = ({ studentId }) => {
  const postNewStudentImage = usePostNewStudentImage(studentId)
  const images = useGetStudentImages(studentId)

  const handleImageUpload = async (event: ChangeEvent<HTMLInputElement>) => {
    const selectedImage = event.target.files?.[0]
    if (selectedImage) {
      try {
        await postNewStudentImage.mutateAsync(selectedImage)
      } catch (e) {
        Sentry.captureException(e)
      }
    }
  }

  return (
    <Box>
      <Label px={3} pt={3} pb={[3, 2]}>
        <input
          type="file"
          accept="image/*"
          style={{ display: "none" }}
          disabled={postNewStudentImage.isLoading}
          onChange={handleImageUpload}
        />
        <Button
          as="div"
          disabled={postNewStudentImage.isLoading}
          sx={{ width: ["100%", "auto"] }}
        >
          {postNewStudentImage.isLoading && <LoadingIndicator />}
          <Icon as={PlusIcon} mr={2} fill="onPrimary" />
          <Trans>Upload Photo</Trans>
        </Button>
      </Label>

      <Flex px={[2, 2]} sx={{ width: "100%", flexWrap: "wrap" }}>
        {images.data?.map((image) => (
          <ImageItem key={image.id} studentId={studentId} image={image} />
        ))}
      </Flex>
    </Box>
  )
}

const fading = keyframes`
  0% {
    background-color: rgba(165, 165, 165, 0.05);
  }
  50% {
    background-color: rgba(165, 165, 165, 0.2);
  }
  100% {
    background-color: rgba(165, 165, 165, 0.05);
  }
`
// TODO: These are some ugly css, might be inconsistent on some devices
//  due to the calc and decimal points, revisit later.
const ImageItem: FC<{ studentId: string; image: StudentImage }> = ({
  studentId,
  image,
}) => (
  <Box p={[1, 2]} sx={{ width: ["33.333%", "25%", "20%", "14.285%"] }}>
    <Link
      to={STUDENT_IMAGE_URL(studentId, image.id)}
      sx={{
        display: "block",
        animation: `1s ease-in-out 0s infinite ${fading}`,
      }}
    >
      <Box pt="100%" sx={{ width: "100%", position: "relative" }}>
        <Image
          loading="lazy"
          src={image.thumbnailUrl}
          sx={{
            backgroundColor: "surface",
            position: "absolute",
            top: 0,
            bottom: 0,
            left: 0,
            right: 0,
            cursor: "pointer",
          }}
        />
      </Box>
    </Link>
  </Box>
)

const VideosView: FC<{ studentId: string }> = ({ studentId }) => {
  const videos = useGetVideos(studentId)
  const postCreateUploadLink = useUploadStudentVideo(studentId)

  const handleImageUpload = async (event: ChangeEvent<HTMLInputElement>) => {
    const selectedVideo = event.target.files?.[0]
    if (selectedVideo) {
      try {
        await postCreateUploadLink.mutateAsync(selectedVideo)
      } catch (e) {
        Sentry.captureException(e)
      }
    }
  }

  return (
    <div>
      <Label px={3} pt={3} pb={[3, 2]}>
        <input
          type="file"
          accept="video/*"
          style={{ display: "none" }}
          disabled={postCreateUploadLink.isLoading}
          onChange={handleImageUpload}
        />
        <Button
          as="div"
          disabled={postCreateUploadLink.isLoading}
          sx={{ width: ["100%", "auto"] }}
        >
          {postCreateUploadLink.isLoading && <LoadingIndicator />}
          <Icon as={PlusIcon} mr={2} fill="onPrimary" />
          <Trans>Upload Video</Trans>
        </Button>
      </Label>

      <Flex px={[2, 2]} sx={{ width: "100%", flexWrap: "wrap" }}>
        {videos.data?.map((video) => (
          <VideoItem
            key={video.id}
            studentId={studentId}
            thumbnailUrl={video.thumbnailUrl}
            playbackUrl={video.playbackUrl}
          />
        ))}
      </Flex>
    </div>
  )
}

const VideoItem: FC<{
  studentId: string
  playbackUrl: string
  thumbnailUrl: string
}> = ({ thumbnailUrl, studentId, playbackUrl }) => {
  const videoDialog = useVisibilityState()

  return (
    <Fragment>
      <Box p={[1, 2]} sx={{ width: ["33.333%", "25%", "20%", "14.285%"] }}>
        <Box
          sx={{
            display: "block",
            animation: `1s ease-in-out 0s infinite ${fading}`,
          }}
          onClick={videoDialog.show}
        >
          <Box
            pt="100%"
            sx={{
              width: "100%",
              position: "relative",
              animation: `1s ease-in-out 0s infinite ${fading}`,
            }}
          >
            <Image
              loading="lazy"
              src={`${thumbnailUrl}`}
              sx={{
                backgroundColor: "surface",
                position: "absolute",
                top: 0,
                bottom: 0,
                left: 0,
                right: 0,
                cursor: "pointer",
              }}
            />
          </Box>
        </Box>
      </Box>
      {videoDialog.visible && (
        <VideoPlayerDialogs
          studentId={studentId}
          src={playbackUrl}
          onClose={videoDialog.hide}
        />
      )}
    </Fragment>
  )
}

const VideoPlayerDialogs: FC<{
  studentId: string
  src: string
  onClose: () => void
}> = ({ src, studentId, onClose }) => {
  const student = useGetStudent(studentId)

  return (
    <Dialog sx={{ maxWidth: ["maxWidth.sm", "maxWidth.lg"] }}>
      <Flex sx={{ alignItems: "center" }}>
        <Typography.Body p={3} sx={{ fontWeight: "bold" }}>
          {student.data?.name || ""}
        </Typography.Body>
        <Button variant="secondary" ml="auto" px={2} mr={3} onClick={onClose}>
          <Icon as={CloseIcon} />
        </Button>
      </Flex>
      <Box sx={{ height: ["100vh", "auto"] }}>
        <Suspense
          fallback={
            <LoadingPlaceholder sx={{ width: "100%", pt: [0, "42.8571%"] }} />
          }
        >
          <LazyVideoPlayer src={src} />
        </Suspense>
      </Box>
    </Dialog>
  )
}

export default PageGallery
