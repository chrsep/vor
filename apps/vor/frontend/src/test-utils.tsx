/* eslint-disable import/no-extraneous-dependencies */
import React, { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { setupI18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import Theme from "./gatsby-plugin-theme-ui"
import enCatalog from "../i18n/lingui/en/messages.js"

const i18n = setupI18n()
// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()

const AllProviders: FC = ({ children }) => {
  // TODO: Maybe this better be outside
  const history = createHistory(createMemorySource("/"))
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
