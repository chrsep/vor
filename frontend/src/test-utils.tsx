/* eslint-disable import/no-extraneous-dependencies */
import React, { FC, ReactElement } from "react"
import { ThemeProvider } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import Theme from "./gatsby-plugin-theme-ui"
import Box from "./components/Box/Box"

const AllProviders: FC = ({ children }) => {
  // TODO: Maybe this better be outside
  const history = createHistory(createMemorySource("/"))
  return (
    <LocationProvider history={history}>
      <ThemeProvider theme={Theme}>
        <Box fontSize={[16, 20]}>{children}</Box>
      </ThemeProvider>
    </LocationProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
