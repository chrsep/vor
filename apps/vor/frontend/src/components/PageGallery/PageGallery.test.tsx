import React from "react"
import { render } from "../../test-utils"
import PageGallery from "./PageGallery"

describe("PageGallery", () => {
  it("should render correctly", () => {
    const { container } = render(<PageGallery />)
    expect(container).toMatchSnapshot()
  })
})
