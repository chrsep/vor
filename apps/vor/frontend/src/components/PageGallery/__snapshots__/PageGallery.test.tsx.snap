// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageGallery should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: var(--theme-ui-colors-surfaceBlurNonTransparent, rgba(255,255,255));
  -webkit-position: sticky;
  position: sticky;
  top: -48px;
  z-index: 10;
}

@supports (backdrop-filter: blur(20px)) {
  .emotion-2 {
    background-color: var(--theme-ui-colors-surfaceBlurTransparent, rgba(255,255,255,0.9));
    -webkit-backdrop-filter: saturate(180%) blur(20px);
    backdrop-filter: saturate(180%) blur(20px);
    -webkit-transform: translateZ(0);
    -moz-transform: translateZ(0);
    -ms-transform: translateZ(0);
    transform: translateZ(0);
    will-change: backdrop-filter;
  }
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-4 {
  margin: 8px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary, #00e399);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary, #027c5b);
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border, rgba(0,0,0,.1));
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  border: none;
  padding: 4px;
}

.emotion-5:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-5:hover,
.emotion-5:focus {
  background-color: var(--theme-ui-colors-primaryLighter, rgba(0,227,153,0.12));
  border-color: var(--theme-ui-colors-primary, #00e399);
  outline: none;
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-icon, rgba(0,0,0,0.6));
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
  fill: var(--theme-ui-colors-textMediumEmphasis, rgba(0,0,0, 0.60));
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow-x: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-8 {
  display: inline-block;
  font-size: 12px;
  color: var(--theme-ui-colors-textMediumEmphasis, rgba(0,0,0, 0.60));
  white-space: nowrap;
}

.emotion-8:last-child {
  color: var(--theme-ui-colors-textPrimary, #027c5b);
}

.emotion-8:last-child>.splitter {
  display: none;
}

.emotion-9:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-10 {
  margin-left: 4px;
  margin-right: 4px;
}

.emotion-14 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: inline-block;
  font-size: 12px;
  color: var(--theme-ui-colors-textMediumEmphasis, rgba(0,0,0, 0.60));
  white-space: nowrap;
}

.emotion-14:last-child {
  color: var(--theme-ui-colors-textPrimary, #027c5b);
}

.emotion-14:last-child>.splitter {
  display: none;
}

.emotion-15 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: relative;
  overflow-x: auto;
}

.emotion-16 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 0;
  min-width: 100%;
  width: 100%;
  -webkit-box-pack: space-around;
  -ms-flex-pack: space-around;
  -webkit-justify-content: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-17 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text, #37383c);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  padding-left: 4px;
  padding-right: 4px;
  margin-bottom: 0;
  color: var(--theme-ui-colors-textPrimary, #027c5b);
  font-size: 14px;
  margin-left: 16px;
  margin-right: 16px;
  white-space: nowrap;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 12px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  z-index: 10;
}

@media screen and (min-width: 40em) {
  .emotion-17 {
    font-size: 14px;
  }
}

.emotion-17:after {
  background-color: var(--theme-ui-colors-textPrimary, #027c5b);
  border-radius: 2px 2px 0 0;
  width: 125%;
  height: 3px;
  content: '';
  margin-top: 8px;
  -webkit-transition: width 100ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: width 100ms cubic-bezier(0.0, 0.0, 0.2, 1);
}

.emotion-18 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text, #37383c);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  padding-left: 4px;
  padding-right: 4px;
  margin-bottom: 0;
  color: var(--theme-ui-colors-textMediumEmphasis, rgba(0,0,0, 0.60));
  font-size: 14px;
  margin-left: 16px;
  margin-right: 16px;
  white-space: nowrap;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 12px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  z-index: 10;
}

@media screen and (min-width: 40em) {
  .emotion-18 {
    font-size: 14px;
  }
}

.emotion-18:after {
  background-color: var(--theme-ui-colors-textPrimary, #027c5b);
  border-radius: 2px 2px 0 0;
  width: 0%;
  height: 3px;
  content: '';
  margin-top: 8px;
  -webkit-transition: width 100ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: width 100ms cubic-bezier(0.0, 0.0, 0.2, 1);
}

.emotion-19 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  height: 1px;
  width: 100%;
  background-color: var(--theme-ui-colors-border, rgba(0,0,0,.1));
  position: absolute;
  bottom: 0;
}

.emotion-21 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-22 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: var(--theme-ui-colors-textMediumEmphasis, rgba(0,0,0, 0.60));
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 16px;
  padding-bottom: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-22 {
    padding-bottom: 8px;
  }
}

.emotion-23 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary, #00e399);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-onPrimary, #000);
  background-color: var(--theme-ui-colors-primary, #00e399);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: rgba(27, 31, 35, 0.1) 0px 1px 0px 0px,rgba(255, 255, 255, 0.03) 0px 1px 0px 0px inset;
  border-color: var(--theme-ui-colors-border, rgba(0,0,0,.1));
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  width: 100%;
}

.emotion-23:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-23:hover,
.emotion-23:focus {
  background-color: var(--theme-ui-colors-primaryDark, #00a06d);
  outline: none;
}

.emotion-23:hover:disabled,
.emotion-23:focus:disabled {
  background-color: var(--theme-ui-colors-primary, #00e399);
}

@media screen and (min-width: 40em) {
  .emotion-23 {
    width: auto;
  }
}

.emotion-24 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  margin-right: 8px;
  fill: var(--theme-ui-colors-onPrimary, #000);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
}

.emotion-25 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 8px;
  padding-right: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

@media screen and (min-width: 40em) {
  .emotion-25 {
    padding-left: 8px;
    padding-right: 8px;
  }
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-3"
        >
          <a
            class="emotion-4"
            data-cy="back-button"
            href="/dashboard/students/overview?studentId=undefined"
          >
            <button
              class="emotion-5"
            >
              <div
                class="emotion-6"
              />
            </button>
          </a>
          <div
            class="emotion-7"
          >
            <a
              class="emotion-8"
              href="/dashboard/students"
            >
              <span
                class="emotion-9"
              >
                Students
              </span>
              <span
                class="splitter emotion-10"
              >
                /
              </span>
            </a>
            <a
              class="emotion-8"
              href="/dashboard/students/overview?studentId=undefined"
            >
              <span
                class="emotion-9"
              />
              <span
                class="splitter emotion-10"
              >
                /
              </span>
            </a>
            <div
              class="emotion-14"
            >
              Gallery
              <span
                class="splitter"
              >
                /
              </span>
            </div>
          </div>
        </div>
        <div
          class="emotion-15"
        >
          <div
            class="emotion-16"
          >
            <p
              class="emotion-17"
            >
              Photos
            </p>
            <p
              class="emotion-18"
            >
              Videos
            </p>
          </div>
          <div
            class="emotion-19"
          />
        </div>
      </div>
      <div
        class="emotion-1"
      >
        <div
          class="emotion-21"
        >
          <label
            class="emotion-22"
          >
            <input
              accept="image/*"
              style="display: none;"
              type="file"
            />
            <div
              class="emotion-23"
            >
              <div
                class="emotion-24"
              />
              Upload Photo
            </div>
          </label>
        </div>
        <div
          class="emotion-25"
        />
      </div>
    </div>
  </div>
</div>
`;
